aspnetcore/performance/caching/memory.md

```diff
@@ -1,164 +1,111 @@
-
-Use the `Set` method to write to the cache. `Set` accepts the key to use to look up the value, the value to be cached, and a set of `MemoryCacheEntryOptions`. The `MemoryCacheEntryOptions` allow you to specify absolute or sliding time-based cache expiration, caching priority, callbacks, and dependencies. These options are detailed below.
-
-The sample code (shown below) uses the `SetAbsoluteExpiration` method on `MemoryCacheEntryOptions` to cache greetings for one minute.
-
-[!code-csharp[Main](memory/sample/src/CachingSample/Middleware/GreetingMiddleware.cs?highlight=7,10,16-18&range=30-58)]
-
-In addition to setting an absolute expiration, a sliding expiration can be used to keep frequently requested items in the cache:
-
-```csharp
-// keep item in cache as long as it is requested at least
-// once every 5 minutes
-new MemoryCacheEntryOptions()
-  .SetSlidingExpiration(TimeSpan.FromMinutes(5))
-```
-
-To avoid having frequently-accessed cache entries growing too stale (because their sliding expiration is constantly reset), you can combine absolute and sliding expirations:
-
-```csharp
-// keep item in cache as long as it is requested at least
-// once every 5 minutes...
-// but in any case make sure to refresh it every hour
-new MemoryCacheEntryOptions()
-  .SetSlidingExpiration(TimeSpan.FromMinutes(5))
-  .SetAbsoluteExpiration(TimeSpan.FromHours(1))
-```
-
-By default, an instance of `MemoryCache` will automatically manage the items stored, removing entries when necessary in response to memory pressure in the app. You can influence the way cache entries are managed by setting their `CacheItemPriority` when adding the item to the cache. For instance, if you have an item you want to keep in the cache unless you explicitly remove it, you would use the `NeverRemove` priority option:
-
-```csharp
-// keep item in cache indefinitely unless explicitly removed
-new MemoryCacheEntryOptions()
-  .SetPriority(CacheItemPriority.NeverRemove))
-```
+Request the `IMemoryCache` instance in the constructor:
 
-When you do want to explicitly remove an item from the cache, you can do so easily using the `Remove` method:
+```c#
+  public class HomeController : Controller
+  {
 
-```csharp
-cache.Remove(cacheKey);
+      public HomeController(IMemoryCache memoryCache)
+      {
+          _memoryCache = memoryCache;
+      }
 ```
 
-## Cache Dependencies and Callbacks
+`IMemoryCache` requires NuGet package "Microsoft.Extensions.Caching.Memory".
 
-You can configure cache entries to depend on other cache entries, the file system, or programmatic tokens, evicting the entry in response to changes. You can register a callback, which will run when a cache item is evicted.
+The following code uses [TryGetValue](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache) to cache and fetch the current time:
```

Comment:
Also maybe link to the specific api for `TryGetValue`? Something like: https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache#Microsoft_Extensions_Caching_Memory_IMemoryCache_TryGetValue_System_Object_System_Object__ instead of https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.caching.memory.imemorycache.

Similarly for the other apis explicitly linked.